### PR TITLE
Full screenshots

### PR DESCRIPTION
**Pull request by Areyana**

### CODE REVIEW

**Comment by AzamatCherchesov** [comment]
#18 Linked issue

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/ExternalScreenshotMaker.kt [hunk 1]
```diff
@@ -23,4 +23,8 @@ class ExternalScreenshotMaker(
     override fun takeScreenshot(file: File) {
         device.takeScreenshot(file, scale, params.quality)
     }
+
+    override fun takeFullScreenshot(file: File) {
```
Comment (AzamatCherchesov) [inline kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/ExternalScreenshotMaker.kt:27]: @Areyana I find this solution not good enough. There are no full view screenshots here. We can make this method unimplemented, add full view implementation for this class(may be not possible), or add some comments/TODO here.

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/Screenshots.kt [hunk 2]
```diff
@@ -16,6 +16,15 @@ interface Screenshots {
...
     fun take(tag: String)
 
+    /**
+     * Takes full screenshot if it is possible.
+     *
+     * Required Permissions: WRITE_EXTERNAL_STORAGE.
+     *
+     * @param tag a unique tag to further identify the screenshot. Must match [a-zA-Z0-9_-]+.
+     */
+    fun takeFull(tag: String)
```
Comment (AzamatCherchesov) [inline kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/Screenshots.kt:26]: @Areyana could you help me to know the purpose of these methods? If you want the basic screenshot engine in the test to be able to take fullscreen screenshots, this is a good idea, but then we need to have an additional option in Kaspresso Builder.

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/Screenshots.kt [hunk 3]
```diff
@@ -24,4 +33,13 @@ interface Screenshots {
...
     fun takeAndApply(tag: String, block: File.() -> Unit)
+
+    /**
+     * Takes full screenshot if it is possible and applies a function on the file.
+     *
+     * Required Permissions: WRITE_EXTERNAL_STORAGE.
+     *
+     * @param tag a unique tag to further identify the screenshot. Must match [a-zA-Z0-9_-]+.
+     */
+    fun takeFullAndApply(tag: String, block: File.() -> Unit)
```
Comment (AzamatCherchesov) [inline kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/Screenshots.kt:44]: @Areyana I think the word "full" in the names of methods and classes is not clear.What do you think about "fullPicture" or "fullWindow", or "fullScreen", or "fullView"

**Review by AzamatCherchesov — CHANGES_REQUESTED** [review]

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/Screenshots.kt (hunk 3 above)
Comment (Areyana) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/Screenshots.kt:44]: fullScreen i guess fine

**Review by AzamatCherchesov — APPROVED** [review]

**Pull request merged**